src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -91,6 +91,15 @@
     @Query("select pe from ProgrammingExercise pe where pe.buildAndTestStudentSubmissionsAfterDueDate > :#{#dateTime}")
     List<ProgrammingExercise> findAllByBuildAndTestStudentSubmissionsAfterDueDateAfterDate(@Param("dateTime") ZonedDateTime dateTime);
 
+    /**
+     * Returns the programming exercises that have manual assessment enabled and a due date higher than the provided date.
+     *
+     * @param dateTime ZonedDateTime object.
+     * @return List<ProgrammingExercise> (can be empty)
+     */
+    @Query("select pe from ProgrammingExercise pe where pe.assessmentType <> 'AUTOMATIC' and pe.dueDate > :#{#dateTime}")
+    List<ProgrammingExercise> findAllByManualAssessmentAndDueDateAfterDate(@Param("dateTime") ZonedDateTime dateTime);
```

Comment:
The method name states the we filter by manual assessment, but we filter by `<> 'AUTOMATIC'`. Is there a reason why don't filter by 'MANUAL'?

Reply:
Because we use `SEMIAUTOMATIC` for that _currently_. I wanted to be safe here and return all exercises that involve some sort of manual assessment. As far as I can see, `MANUAL` is unused currently for the programming exercises.

Reply:
Thanks, seems reasonable.